### DISCUSSION TITLE
Best way to disable line numbers

### DISCUSSION THREAD

**Discussion by valignatev** [orbitalquark/textadept#115] [Q&A] [answered]
Hey, just found out about Textadept and playing around with init.lua. Based on scrolling through the API, this is how I was able to disable line numbers:

```lua
view.margin_width_n[view.MARGIN_NUMBER] = 0
```

Is this an idiomatic way of doing it? If not - how would you do it the right way?

UPD: oh nevermind, apparently it only works interactively and does nothing upon reloading the editor

It sorta works with 
```lua
events.connect(events.FILE_OPENED, function()
  view.margin_width_n[1] = 0
end)
```
but only for newly opened files, and the settings file still has line numbers

UPD UPD:

This is the final version I came up with:
```lua
-- Disable line numbers.
disable_line_numbers_upon = {'INITIALIZED', 'FILE_OPENED', 'VIEW_NEW', 'BUFFER_NEW'}

for i, event in ipairs(disable_line_numbers_upon) do
  events.connect(events[event], function()
    view.margin_width_n[view.MARGIN_NUMBER] = 0
  end)
end
```

**Comment by orbitalquark** [comment, marked as answer]
That's the way I do it. You can make it a function and use a key binding or menu item to toggle hiding/showing it if you save the original width first.

**Reply by valignatev** [reply]
Thanks!
